cmake/compiler.cmake

```diff
@@ -211,6 +211,13 @@ option(ENABLE_WERROR "Make all compiler warnings into errors" OFF)
 
 option(ENABLE_UB_SANITIZER "Make the compiler generate runtime code to perform undefined behaviour checks" OFF)
 
+option(ENABLE_HARDENING "Enable compiler options that harden against memory corruption attacks" ON)
+if (ENABLE_HARDENING)
+    set(HARDENING_FLAGS "-fstack-protector")
```

Comment:
Shouldn't we use `-fstack-protector-strong` and require gcc 4.9+ when handening is enabled? (I lean on description of those options from the [Debian's hardening page](https://wiki.debian.org/Hardening#DEB_BUILD_HARDENING_STACKPROTECTOR_.28gcc.2Fg.2B-.2B-_-fstack-protector-strong.29).)

Don't know clang's version, where it appears, but we can do the similar version check for clang. So if one attempts to build the harden executable using an old compiler, he/she will get an explicit error.

Reply:
I've added all Debian handening options. [cbench](https://github.com/tarantool/cbench/) and [tuple.perftest](https://github.com/tarantool/tarantool/blob/master/perf/tuple.cc) don't catch any performance regression. full-ci testing passed, so I guess gcc version check is not required.